dhcpv4/server4/server.go

```diff
@@ -117,10 +128,13 @@ func NewServer(addr *net.UDPAddr, handler Handler, opt ...ServerOpt) (*Server, e
 	}
 	if s.conn == nil {
 		var err error
-		s.conn, err = net.ListenUDP("udp4", addr)
+		conn, err := net.ListenUDP("udp4", addr)
 		if err != nil {
 			return nil, err
 		}
+		s.mu.Lock()
```

Comment:
Why is there a connection mutex at all? You aren't able to change it at runtime

Reply:
I added it because there was a race condition when I added the check in `Close` for a nil conn (in case someone calls `Close` twice), but I will remove it

Reply:
sounds like the same thing that you fixed in https://github.com/insomniacslk/dhcp/pull/274 ? Except that there was no reason here until I changed how `Close` works :D (which I reverted for now)